inbc-program/inbc/command/command.py

```diff
@@ -129,7 +129,7 @@ def invoke_update(self, args: Any) -> None:
 
         @param args: arguments passed to command-line tool.
         """
-        super()._send_manifest(args, RESTART_CHANNEL)
+        super()._send_manifest(args, RESTART_CMD_CHANNEL)
 
     def search_response(self, payload: Any) -> None:
         """Search for keywords in response message
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Maintainability$</h2>**
**Suggestion:** The terminate_operation method in the Command class directly modifies the shared.exit_code and shared.running flags. This tightly couples the Command class to the shared module and can make the code harder to test and maintain. Consider using a callback or observer pattern to decouple these concerns.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
def terminate_operation(self, status: str, return_code: int) -> None:
    if status == COMMAND_SUCCESS:
        print(f"\n {self._cmd_type.upper()} Command Execution is Completed")
    elif status == COMMAND_FAIL:
        print(f"\n {self._cmd_type.upper()} Command Execution FAILED")
        self.on_failure(abs(return_code))
    logger.info("INBC code: {0}".format(return_code))
    self.on_terminate()
    self._update_timer.stop()

def on_failure(self, return_code: int):
    shared.exit_code = return_code

def on_terminate(self):
    shared.running = False
```